src/contexts/TodoContext.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useMemo, useState } from 'react';
+import { Todos } from '../types/Todos';
+import { Status } from '../types/Status';
+import { filterTodoByStatus } from '../services/FilteringByStatus';
+import { ContextProps } from '../types/ContextProps';
+import { useLocalStorage } from '../hooks/LocalStorage';
+
+export const TodoContext = React.createContext<ContextProps>({
+  todos: [],
+  addTodo: () => { },
+  deleteTodo: () => { },
+  setTodos: () => { },
+  filterTodoByStatus: () => [],
+  setStatus: () => [],
+  status: Status.All,
+  clearTodo: () => [],
+  editTodo: () => [],
+  toggleAll: () => {},
+});
+
+type Childer = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<Childer> = ({ children }) => {
+  const [todos, setTodos] = useLocalStorage<Todos[]>('todos', []);
+  const [status, setStatus] = useState(Status.All);
+
+  const addTodo = (todo: Todos) => {
+    setTodos(prevTodo => [
+      ...prevTodo,
+      todo,
+    ]);
+  };
+
+  const deleteTodo = (id: number) => {
+    setTodos((prevTodos) => prevTodos.filter((todo) => todo.id !== id));
+  };
+
+  const clearTodo = () => {
+    setTodos(currentTodos => currentTodos.filter(todo => !todo.completed));
+  };
+
+  const editTodo = (id: number, editTitle: string) => {
+    setTodos(currentTodos => currentTodos
+      .map(todo => (todo.id === id
+        ? {
+          ...todo,
+          title: editTitle,
+        }
+        : todo)));
```

Comment:
it is better to not call setTodos, if todo.title the same as before, I mean, if user tried to change title, but didn't change it - don't call set state.